cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -119,8 +186,168 @@ pub fn maybe_make_service_config_file(
     Ok(service_config_path)
 }
 
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+struct ProcessState {
+    name: String,
+    namespace: String,
+    status: String,
+    system_time: String,
+    age: u64,
+    is_ready: String,
+    restarts: u64,
+    exit_code: i32,
+    pid: u64,
+    #[serde(rename = "IsRunning")]
+    is_running: bool,
+}
+
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+pub struct ProcessStates(Vec<ProcessState>);
+
+impl ProcessStates {
+    pub fn read(socket: impl AsRef<Path>) -> Result<ProcessStates, ServiceError> {
+        let mut cmd = base_process_compose_command(socket.as_ref());
+        let output = cmd
+            .arg("list")
+            .args(["--output", "json"])
+            .output()
+            .map_err(ServiceError::ProcessComposeCmd)?;
+        if !output.status.success() {
+            return Err(ServiceError::from_output(String::from_utf8_lossy(
+                &output.stderr,
+            )));
+        }
+        let mut processes: ProcessStates =
+            serde_json::from_slice(&output.stdout).map_err(ServiceError::ParseOutput)?;
+        processes
+            .0
+            .retain(|state| state.name != PROCESS_NEVER_EXIT_NAME);
+
+        Ok(processes)
+    }
+
+    pub fn get_running_names(&self) -> Vec<String> {
+        self.0
+            .iter()
+            .filter(|state| state.is_running)
+            .map(|state| state.name.clone())
+            .collect()
+    }
```

Comment:
nit: conventionally, rust [doesn't use `get_` prefixes](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter)

suggestion: add a doc comment
```suggestion
    /// Get the names of the currently running processes
    pub fn get_running_names(&self) -> Vec<String> {
        self.0
            .iter()
            .filter(|state| state.is_running)
            .map(|state| state.name.clone())
            .collect()
    }
```


suggesetion nb: based on how this is used, if we dont need need owned strings downstream anyway, i think we could get away without cloning by returning an `impl Iterator<Item = &str>`,

Reply:
Fixed